src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,468 @@
+    };
+
+    this._videoTracks = [];
+    if (mediaElement.videoTracks != null) {
+      this._videoTracks = createVideoTracks();
+      mediaElement.videoTracks.onaddtrack = () => {
+        const newVideoTracks = createVideoTracks();
+        if (areTrackArraysDifferent(this._videoTracks, newVideoTracks)) {
+          this._videoTracks = newVideoTracks;
+          this.trigger("availableTracksChange", "video");
+        }
+      };
+      mediaElement.videoTracks.onremovetrack = () => {
+        const newVideoTracks = createVideoTracks();
+        if (areTrackArraysDifferent(this._videoTracks, newVideoTracks)) {
+          this._videoTracks = newVideoTracks;
+          this.trigger("availableTracksChange", "video");
+        }
+      };
+    }
+
+    this._onTrackChange(mediaElement);
+  }
+
+  public setAudioTrackById(id?: string|number): void {
+    for (let i = 0; i < this._audioTracks.length; i++) {
+      const { track, origTrack } = this._audioTracks[i];
+      if (track.id === id) {
+        origTrack.enabled = true;
+        return;
+      }
+    }
+    throw new Error("Audio track not found.");
+  }
+
+  public disableTextTrack(): void {
+    for (let i = 0; i < this._textTracks.length; i++) {
+      const { origTrack } = this._textTracks[i];
+      origTrack.mode = "disabled";
+    }
+  }
+
+  public setTextTrackById(id?: string|number): void {
+    let hasSetTrack = false;
+    for (let i = 0; i < this._textTracks.length; i++) {
+      const { track, origTrack } = this._textTracks[i];
+      if (track.id === id) {
+        origTrack.mode = "showing";
+        hasSetTrack = true;
+      } else if (origTrack.mode === "showing" || origTrack.mode === "hidden") {
+        origTrack.mode = "disabled";
+      }
+    }
+    if (!hasSetTrack) {
+      throw new Error("Text track not found.");
+    }
+  }
+
+  public setVideoTrackById(id?: string): void {
+    for (let i = 0; i < this._videoTracks.length; i++) {
+      const { track, origTrack } = this._videoTracks[i];
+      if (track.id === id) {
+        origTrack.selected = true;
+        return;
+      }
+    }
+    throw new Error("Video track not found.");
+  }
+
+  public getChosenAudioTrack(): ITMAudioTrack|null|undefined {
+    for (let i = 0; i < this._audioTracks.length; i++) {
+      const { track, origTrack } = this._audioTracks[i];
+      if (origTrack.enabled) {
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getChosenTextTrack(): ITMTextTrack|null|undefined {
+    for (let i = 0; i < this._textTracks.length; i++) {
+      const { track, origTrack } = this._textTracks[i];
+      if (origTrack.mode === "showing") {
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getChosenVideoTrack(): ITMVideoTrack|null|undefined {
+    for (let i = 0; i < this._videoTracks.length; i++) {
+      const { track, origTrack } = this._videoTracks[i];
+      if (origTrack.selected) {
+        return track;
+      }
+    }
+    return null;
```

Comment:
So this is never set to `undefined`?

If the API does not exist we should return `undefined`

Reply:
done